modules/BuildTools.Tasks/CreateSourceLink.cs

```diff
@@ -51,20 +51,52 @@ public override bool Execute()
             SourceLinkRoot += '*';
 
             var codeSource = ConvertUrl();
-            var data = new JObject
+
+            // TODO allow multiple. See https://github.com/aspnet/Universe/issues/858
+            var mappings = new List<Mapping>
             {
-                ["documents"] = new JObject
-                {
-                    [SourceLinkRoot] = codeSource
-                }
+                new Mapping { LocalPath = SourceLinkRoot, Url = codeSource },
             };
-            File.WriteAllText(DestinationFile, data.ToString(Formatting.None));
 
+            GenerateSourceLink(DestinationFile, mappings);
             SourceLinkFile = DestinationFile;
-
             return true;
         }
 
+        private static void GenerateSourceLink(string filePath, List<Mapping> mappings)
```

Comment:
Any reason not to just use Json.NET for this? Looks like it's available anyway, no?

Reply:
This dependency on Json.Net was causes build failures when opening solutions in VS 2019. We already fixed this in 2.2, and cherry-picked that change back to 2.1.